File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int i = 1;
        while (i < intervals.size()) {
            if (intervals[i-1][1] >= intervals[i][0]) {
                intervals[i-1][1] = max(intervals[i][1], intervals[i-1][1]);
                intervals.erase(intervals.begin() + i);
            } else {
                i++;
            }
        }
        return intervals;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        i = 1
        while i < len(intervals):
            if intervals[i-1][1] >= intervals[i][0]:
                intervals[i-1][1] = max(intervals[i][1], intervals[i-1][1]) 
                # intervals[i-1][0] = min(intervals[i][0], intervals[i-1][0])
                intervals.pop(i)
                i -= 1
            i += 1
        return intervals
